examples/Baselines/Halite_competition/paddle/train.py

```diff
@@ -0,0 +1,259 @@
+        episode_halite = [0] * player_num
+        episode_ship_halite = [0] * player_num
+        episode_ship_num = [0] * player_num
+        episode_shipyard_num = [0] * player_num
+
+        # record the alive status of players
+        agent_done = [False] * player_num
+
+        # since we train the agent with a random agent, we also save the episode halite obtained by
+        # the random agent and compute a winning rate
+        random_agent_halite = 0
+
+        # reset the parameters in player
+        for ind, board in zip(player_index, boards):
+            players[ind].prepare_train()
+
+        while True:
+
+            total_step += 1
+            episode_step += 1
+
+            # record the actions of agents
+            actions = [None] * player_num
+
+            # sampling actions
+            for ind, board in zip(player_index, boards):
+
+                # compute the action for those alive agents
+                if not agent_done[ind]:
+
+                    actions[ind] = players[ind].take_action(board, "sample")
+
+            # fetch data of each training agent
+            player_infos, terminal = env.step(actions)
+            observation_next = [infos[0] for infos in player_infos]
+            rews = [infos[1] for infos in player_infos]
+            dones = [infos[2] for infos in player_infos]
+            boards_next = [
+                Board(obs_next, configuration) for obs_next in observation_next
+            ]
+
+            # reset the flag of done
+            for ind, done in zip(player_index, dones):
+                agent_done[ind] = done
+
+            boards = boards_next
+
+            if terminal:
+
+                # update the final state of ships and shipyards and save data
+                for ind, board in zip(player_index, boards):
+                    players[ind].update_state(board)
+                    episode_ship_halite[ind] = sum(
+                        [ship.halite for ship in board.current_player.ships])
+                    episode_halite[ind] = board.current_player.halite
+                    episode_ship_num[ind] = len(board.current_player.ship_ids)
+                    episode_shipyard_num[ind] = len(
+                        board.current_player.shipyard_ids)
+
+                random_agent_halite = boards[0].opponents[0].halite
+                break
+
+        # train agents
+        if len(players[0].ship_buffer):
+            value_loss, action_loss, entropy = players[0].train_ship_agent()
+            #logx.msg("ship loss : {0}".format(ship_loss))
+            logx.metric(
+                "train", {
+                    "ship_value_loss": value_loss,
+                    "ship_policy_loss": action_loss,
+                    "ship_entropy": entropy
+                }, total_step)
+
+        # print logs
+        logx.msg("episode:{0}, episode_step:{1}, total_step:{2}".format(
+            episode, episode_step, total_step))
+
+        # snippet of code to test the agent
```

Comment:
We can remove these codes?

Reply:
Yes.  They will be removed in the next commit.